### PR TITLE
doc: migrate two guides from notation repo and update notatation-config-file.md

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Fix https://github.com/notaryproject/notation/issues/191

Signed-off-by: Feynman Zhou <feynmanzhou@microsoft.com>

### CODE REVIEW

File: content/en/docs/tutorials/notation-config-file.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+The default location and file will be stored at: `~/Users/exampleuser/Library/Application Support/notation/config.json`.
+
+## Properties
+
+| Property                  | Type     | Value                                                                                                                                                     |
+| ------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `verificationCerts.certs` | _array_  | collection of name/value pairs for a collection of public certs that are used for verification. These may be replaced with a future policy configuration. |
+| `cert.name`               | _string_ | a named reference to the certificate                                                                                                                      |
+| `cert.path`               | _string_ | a location by which the certificate can be found by the notation cli or notation libraries                                                                |
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/notation-config-file.md:36]: No longer need. Due to TS/TP will be included.

File: content/en/docs/tutorials/notation-config-file.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            {
+                "name": "wabbit-networks",
+                "path": "/home/demo/.config/notation/localkeys/wabbit-networks.crt"
+            },
+            {
+                "name": "import.acme-rockets",
+                "path": "/home/demo/.config/notation/localkeys/import.acme-rockets.crt"
+            }
+        ]
+    },
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/notation-config-file.md:61]: No longer need. Due to TS/TP will be included.

File: content/en/docs/tutorials/notation-config-file.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    "insecureRegistries": [
+        "registry.wabbit-networks.io"
+    ]
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/notation-config-file.md:64]: Also need:
    "credsStore": "pass",
    "credHelpers": {
        "registry.io": "pass"
    },
    "signatureFormat": "jws"

File: content/en/docs/tutorials/notation-config-file.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+| `signingKeys.default`     | _string_ | the signing key to be used when `notation sign` is called without `--name`                                                                                |
+| `signingKeys.keys`        | _array_  | a collection of name/value pairs of signing keys.                                                                                                         |
+| `key.name`                | _string_ | a named reference to the key                                                                                                                              |
+| `key.keyPath`             | _string_ | a location by which the key can be found by the notation cli or notation libraries                                                                        |
+| `key.certPath`            | _string_ | a location by which the paired certificate can be found by the notation cli or notation libraries                                                         |
+| `insecureRegistries`      | _array_  | a list of registries that may be used without https                                                                                                       |
+
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/notation-config-file.md:43]: credsStore: default trust store name
credHelpers: a registry host name address to trust store name map
signatureFormat: define signature envelope format, jws/cose

File: content/en/docs/tutorials/directory-structure.md [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+| `NOTATION_BIN`     | Directory for executable binaries                                                   |
+| `NOTATION_LIBEXEC` | Directory for binaries not meant to be executed directly by users' shell or scripts |
+| `NOTATION_CONFIG`  | Directory for configurations                                                        |
+| `NOTATION_CACHE`   | Directory for user-specific cache                                                   |
+
+Although it is recommended to install `notation` with its plugins and default configurations at the system level, it is possible to install at the user level.
+
+On Unix systems, `notation` follows [Filesystem Hierarchy Standard][FHS] for system level directories and [XDG Base Directory Specification][XDG] for user level directories. On Windows, [Known Folders][KF] and [App Settings][AS] are followed equivalently. On Darwin, [macOS File System][macOS_FS] with [System Integrity Protection][SIP] is followed equivalently. If a file with the same name exists at the system level and the user level, the file at the user level takes over the priority.
+
+### System Level
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/directory-structure.md:35]: RC1 does not support system level

File: content/en/docs/tutorials/directory-structure.md [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+## Category
+
+The directories for various components are classified into the following catagories.
+
+| Alias              | Description                                                                         |
+| ------------------ | ----------------------------------------------------------------------------------- |
+| `NOTATION_BIN`     | Directory for executable binaries                                                   |
+| `NOTATION_LIBEXEC` | Directory for binaries not meant to be executed directly by users' shell or scripts |
+| `NOTATION_CONFIG`  | Directory for configurations                                                        |
+| `NOTATION_CACHE`   | Directory for user-specific cache                                                   |
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/directory-structure.md:29]: cache support was removed.

**Comment by dtzar** [comment]
LGTM I'd just fix issues JeyJeyGao calls out and we can approve/merge.

File: content/en/docs/tutorials/notation-config-file.md (hunk 3 above)
Comment (FeynmanZhou) [reply content/en/docs/tutorials/notation-config-file.md:64]: What's the `pass` mean here?

Comment (JeyJeyGao) [reply content/en/docs/tutorials/notation-config-file.md:56]: This is the list of currently available credentials helpers and where you can download them from:

D-Bus Secret Service: https://github.com/docker/docker-credential-helpers/releases
Apple macOS keychain: https://github.com/docker/docker-credential-helpers/releases
Microsoft Windows Credential Manager: https://github.com/docker/docker-credential-helpers/releases
[pass](https://www.passwordstore.org/): https://github.com/docker/docker-credential-helpers/releases

File: content/en/docs/tutorials/notation-config-file.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+## Samples
+
+See the a sample configuration in a Linux environment: `~/.config/notation/config.json`. 
+
+```json
+{
+    "credsStore": "pass",
+    "credHelpers": {
+    "registry.io": "pass"
+    },
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/notation-config-file.md:53]:
```suggestion
    "credHelpers": {
        "registry.io": "pass"
    },
```

File: content/en/docs/tutorials/notation-config-file.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    "insecureRegistries": [
+        "registry.wabbit-networks.io"
+    ]
+}
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/notation-config-file.md:57]: Do we need `"signatureFormat": "jws"` here?

File: content/en/docs/tutorials/notation-config-file.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+| `signingKeys.default`     | _string_ | the signing key to be used when `notation sign` is called without `--name`                                                                                |
+| `signingKeys.keys`        | _array_  | a collection of name/value pairs of signing keys.                                                                                                         |
+| `key.name`                | _string_ | a named reference to the key                                                                                                                              |
+| `key.keyPath`             | _string_ | a location by which the key can be found by the notation cli or notation libraries                                                                        |
+| `key.certPath`            | _string_ | a location by which the paired certificate can be found by the notation cli or notation libraries                                                         |
+| `insecureRegistries`      | _array_  | a list of registries that may be used without https                                                                                                       |
+
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/notation-config-file.md:43]: Do we need signatureFormat?

File: content/en/docs/tutorials/notation-config-file.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+### Darwin 
+
+The default location and file will be stored at: `~/Users/exampleuser/Library/Application Support/notation/config.json`.
+
+## Properties
+
+| Property                  | Type     | Value                                                                                                                                                     |
+| ------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `credsStore` | _string_  | default trust store name. |
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/notation-config-file.md:34]: Can we format the markdown to get them aligned?

File: content/en/docs/tutorials/notation-config-file.md (hunk 9 above)
Comment (FeynmanZhou) [reply content/en/docs/tutorials/notation-config-file.md:43]: It was added in L36.

File: content/en/docs/tutorials/directory-structure.md [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+
+The directories for various components are classified into the following catagories.
+
+| Alias              | Description                                                                         |
+| ------------------ | ----------------------------------------------------------------------------------- |
+| `NOTATION_BIN`     | Directory for executable binaries                                                   |
+| `NOTATION_LIBEXEC` | Directory for binaries not meant to be executed directly by users' shell or scripts |
+| `NOTATION_CONFIG`  | Directory for configurations                                                        |
+
+Although it is recommended to install `notation` with its plugins and default configurations at the system level, it is possible to install at the user level.
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/directory-structure.md:29]: We do not support system level for Beta.1 and RC.1. Can we remove everything related to system level.

Comment (FeynmanZhou) [reply content/en/docs/tutorials/directory-structure.md]: Good catch. I remove all system-level descriptions from this guide.

File: content/en/docs/tutorials/directory-structure.md [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+        └── AppData
+            ├── Local
+            │   └── notation
+            │       └── signatures
+            │           └── sha256
+            │               └── 05b3abf2579a5eb66403cd78be557fd860633a1fe2103c7642030defe32c657f
+            │                   └── sha256
+            │                       ├── 2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae.sig
+            │                       ├── b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9.sig
+            │                       └── fcde2b2edba56bf408601fb721fe9b5c338d10ee429ea04fae5511b68fbf8fb9.sig
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/directory-structure.md:234]: This is cache directory. It should be delete.

File: content/en/docs/tutorials/notation-config-file.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+> Note: You can find the details of Notation directory structure for system configuration in this [guide]({{< ref "/docs/tutorials/directory-structure" >}}).
+
+### Linux
+
+`config.json` should be mannually created and placed in `~/.config/notation/config.json`. 
+
+### Windows
+
+`config.json` should be mannually created and placed in in `C:/Users/AppData/Roaming/notation/config.json`.
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/notation-config-file.md:22]:
```suggestion
`config.json` should be mannually created and placed in in `C:/Users/exampleuser/AppData/Roaming/notation/config.json`.
```

File: content/en/docs/tutorials/directory-structure.md [hunk 14]
```diff
@@ -0,0 +1,315 @@
...
+│           └── x509
+│               ├── ca
+│               │   ├── acme-rockets
+│               │   │   ├── cert1.pem
+│               │   │   └── cert2.pem
+│               │   └── wabbit-networks
+│               │       └── cert3.pem
+│               └── tsa
+│                   └── publicly-trusted-tsa
+│                       └── tsa-cert1.pem
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/directory-structure.md:163]: system directory need to be deleted.

File: content/en/docs/tutorials/directory-structure.md [hunk 15]
```diff
@@ -0,0 +1,315 @@
...
+│                           └── publicly-trusted-tsa
+│                               └── tsa-cert2.pem
+└── usr
+    ├── bin
+    │   └── notation
+    └── libexec
+        └── notation
+            └── plugins
+                └── com.example.foo
+                    └── notation-com.example.foo
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/directory-structure.md:194]: system directory need to be deleted.

File: content/en/docs/tutorials/directory-structure.md [hunk 16]
```diff
@@ -0,0 +1,315 @@
...
+│           └── x509
+│               ├── ca
+│               │   ├── acme-rockets
+│               │   │   ├── cert1.pem
+│               │   │   └── cert2.pem
+│               │   └── wabbit-networks
+│               │       └── cert3.pem
+│               └── tsa
+│                   └── publicly-trusted-tsa
+│                       └── tsa-cert1.pem
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/directory-structure.md:222]: system directory need to be deleted.

File: content/en/docs/tutorials/directory-structure.md [hunk 17]
```diff
@@ -0,0 +1,315 @@
...
+│               └── x509
+│                   ├── ca
+│                   │   ├── acme-rockets
+│                   │   │   ├── cert1.pem
+│                   │   │   └── cert2.pem
+│                   │   └── wabbit-networks
+│                   │       └── cert3.pem
+│                   └── tsa
+│                       └── publicly-trusted-tsa
+│                           └── tsa-cert1.pem
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/directory-structure.md:268]: system directory need to be deleted.

File: content/en/docs/tutorials/directory-structure.md [hunk 18]
```diff
@@ -0,0 +1,315 @@
...
+│           
+└── usr
+    └── local
+        ├── bin
+        │   └── notation
+        └── lib
+            └── notation
+                └── plugins
+                    └── com.example.foo
+                        └── notation-com.example.foo
```
Comment (JeyJeyGao) [inline content/en/docs/tutorials/directory-structure.md:302]: system directory need to be deleted.

**Review by JeyJeyGao — APPROVED** [review]
LGTM

File: content/en/docs/tutorials/directory-structure.md [hunk 19]
```diff
@@ -0,0 +1,245 @@
...
+
+## Examples
+
+Examples are shown on various operating systems where the user `exampleuser` overrides the `notation` config and the trust policy.
+
+### Unix
+
+```
+/
+├── home
```
Comment (shizhMSFT) [inline content/en/docs/tutorials/directory-structure.md:149]: Should it be
```suggestion
└── home
```
?

This comment also applies to the Darwin section.

File: content/en/docs/tutorials/directory-structure.md [hunk 20]
```diff
@@ -0,0 +1,245 @@
...
+                        └── x509
+                            ├── ca
+                            │   └── acme-rockets
+                            │       └── cert4.pem
+                            └── tsa
+                                └── publicly-trusted-tsa
+                                    └── tsa-cert2.pem
+```
+
+### Darwin
```
Comment (shizhMSFT) [inline content/en/docs/tutorials/directory-structure.md:204]: Do you want to mention `macOS`?

File: content/en/docs/tutorials/notation-config-file.md [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+
+`config.json` should be manually created and placed in `~/.config/notation/config.json`. 
+
+### Windows
+
+`config.json` should be manually created and placed in in `C:/Users/exampleuser/AppData/Roaming/notation/config.json`.
+
+### Darwin 
+
+`config.json` should be manually created and placed in in `~/Users/exampleuser/Library/Application Support/notation/config.json`.
```
Comment (shizhMSFT) [inline content/en/docs/tutorials/notation-config-file.md:26]:
```suggestion
`config.json` should be manually created and placed in in `/Users/exampleuser/Library/Application Support/notation/config.json`.
```
or
```suggestion
`config.json` should be manually created and placed in in `~/Library/Application Support/notation/config.json`.
```

File: content/en/docs/tutorials/notation-config-file.md [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+
+> Note: You can find the details of Notation directory structure for system configuration in this [guide]({{< ref "/docs/tutorials/directory-structure" >}}).
+
+### Linux
+
+`config.json` should be manually created and placed in `~/.config/notation/config.json`. 
+
+### Windows
+
+`config.json` should be manually created and placed in in `C:/Users/exampleuser/AppData/Roaming/notation/config.json`.
```
Comment (shizhMSFT) [inline content/en/docs/tutorials/notation-config-file.md:22]: To be consistent with other OS,

In PowerShell
```suggestion
`config.json` should be manually created and placed in in `~/AppData/Roaming/notation/config.json`.
```

or in CMD
```suggestion
`%HOMEPATH%/AppData/Roaming/notation/config.json`.
```

File: content/en/docs/tutorials/directory-structure.md (hunk 20 above)
Comment (FeynmanZhou) [reply content/en/docs/tutorials/directory-structure.md:204]: How about using `Darwin / macOS` here?

**Review by dtzar — DISMISSED** [review]

**Review by dtzar — APPROVED** [review]

**Comment by FeynmanZhou** [comment]
I will submit a PR to remove the duplicated guides in https://github.com/notaryproject/notation/tree/main/specs

**Pull request merged**